### PR TITLE
Implement reconnections

### PR DESCRIPTION
**Pull request by ernestojpg**
This pull request implements #108 , allowing the service to recover from temporary database outages, such as those which occur during a database restart or brief loss of network connectivity.

Because the underlying library from mauricio does not support this functionality yet, I have implemented the reconnection logic in our AsyncConnectionPool.createConnection() method, just using a counter and Vert.x timers.

Because testing problems connecting to the database is not easy using real databases, I have used [Mockito](http://site.mockito.org/) for the tests.

Regards.

### CODE REVIEW

File: src/main/java/io/vertx/ext/asyncsql/impl/BaseSQLClient.java [hunk 1]
```diff
@@ -49,15 +49,11 @@
   protected final Logger log = LoggerFactory.getLogger(this.getClass());
   protected final Vertx vertx;
 
-  protected int maxPoolSize;
-  protected int transactionTimeout;
```
Comment (ernestojpg) [inline src/main/java/io/vertx/ext/asyncsql/impl/BaseSQLClient.java]: `transactionTimeout ` was not used

File: src/main/java/io/vertx/ext/asyncsql/impl/BaseSQLClient.java [hunk 2]
```diff
@@ -49,15 +49,11 @@
...
-  protected String registerAddress;
```
Comment (ernestojpg) [inline src/main/java/io/vertx/ext/asyncsql/impl/BaseSQLClient.java]: `registerAddress ` was not used

File: src/main/asciidoc/groovy/index.adoc [hunk 3]
```diff
@@ -228,7 +254,11 @@ Both the PostgreSql and MySql clients take the same configuration:
   "password" : <your-password>,
   "database" : <name-of-your-database>,
   "charset" : <name-of-the-character-set>,
+  "connectTimeout" : <timeout-in-milliseconds>,
+  "testTimeout" : <timeout-in-milliseconds>,
```
Comment (ernestojpg) [inline src/main/asciidoc/groovy/index.adoc]: These two properties were missing in the documentation.

**Comment by Narigo** [comment]
Thanks again @ernestojpg ! :)

**Pull request merged**